Rules/ReviewUnusedParameter.cs

```diff
@@ -51,6 +51,29 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
                     continue;
                 }
 
+                // bail out if $MyInvocation.BoundParameters used
+                if (variableCount.ContainsKey("MyInvocation"))
+                {
+                    bool isBound = false;
+                    IEnumerable<Ast> memExpAst = scriptBlockAst.FindAll(oneAst => oneAst is MemberExpressionAst, false);
+                    foreach (MemberExpressionAst memberExpressionAst in memExpAst)
+                    {                        
+                        if (memberExpressionAst.Expression is VariableExpressionAst &&

```

Comment:
Use the nicer pattern matching syntax that allows you to conditionally create a new typed local as part of the `is` condition - https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#pattern-matching.

```
memberExpressionAst.Expression is VariableExpressionAst veAst &&
veAst.VariablePath.UserPath...
```

Reply:
Nice, thanks for the suggestion